pallets/creditcoin/src/benchmarking.rs

```diff
@@ -334,6 +334,17 @@ benchmarks! {
 		let root = RawOrigin::Root;
 		let contract = GCreContract::default();
 	}: _(root, contract)
+
+	register_address_v2 {
+		let who: T::AccountId = lender_account::<T>(false);
+		let ktypeid = KeyTypeId(*b"dumy");
+		let seed = "//who".as_bytes().to_vec();
+		let pkey = ecdsa_generate(ktypeid, Some(seed));
+		let address = EVMAddress::from_public(&pkey);
+		let message = sp_io::hashing::sha2_256(who.encode().as_slice());
+		let signature = ecdsa_sign(ktypeid, &pkey, &message).expect("ecdsa signature");
+		let proof = OwnershipProof::EthSign(signature);
```

Comment:
If I'm being super pedantic, ideally we would have separate benchmarks for the two signature types (which could have different computational costs): `register_address_v2_eth_sign` and `register_address_v2_personal_sign`.

Then in the weight annotation you'd have a match on the ownership proof:

```rust
#[pallet::weight(match ownership_proof {
    OwnershipProof::EthSign(_) => <T as Config>::WeightInfo::register_address_v2_eth_sign(),
    OwnershipProof::PersonalSign(_) => <T as Config>::WeightInfo::register_address_v2_personal_sign(),
  }]
```

Reply:
I think this proposal will break `scripts/check-parity-bw-extrinsics-benchmarks-weights.sh`, otherwise sounds reasonable to me.